lib/plugins/haskell.js

```diff
@@ -0,0 +1,28 @@
+import { HASKELL_IMPORT } from '../../packages/helper-grammar-regex-collection/index.js';
+import insertLink from '../insert-link';
+import preset from '../pattern-preset';
+
+export default class Haskell {
+  static resolve({ path, target }) {
+    const [, user, repo] = path.split('/');
+    const filePath = target.replace(/\./g, '/');
+    return [
+      `{BASE_URL}/${user}/${repo}/blob/master/src/${filePath}.hs`,
+      `{BASE_URL}/${user}/${repo}/blob/master/lib/${filePath}.hs`,
+      `{BASE_URL}/${user}/${repo}/blob/master/${filePath}.hs`,
+      `https://hackage.haskell.org/package/base-4.9.1.0/docs/${target.replace(/\./g, '-')}.html`,
```

Comment:
I think this can be replaced by `https://hackage.haskell.org/package/base/docs/${target.replace(/\./g, '-')}.html`, which will redirect to the `base-4.9.1.0` url, but would presumably stay up-to-date. For example: https://hackage.haskell.org/package/base/docs/Data-List.html